internal/storage/hub/local_source.go

```diff
@@ -39,19 +41,36 @@ func NewLocalSourceFromConf(_ context.Context, conf *Conf) (*LocalSource, error)
 		return nil, err
 	}
 
-	return NewLocalSource(LocalParams{
-		BundlePath: conf.Local.BundlePath,
-		SecretKey:  conf.Credentials.WorkspaceSecret,
-		TempDir:    conf.Local.TempDir,
-		CacheSize:  conf.CacheSize,
-	})
+	lp := LocalParams{
+		BundlePath:    conf.Local.BundlePath,
+		TempDir:       conf.Local.TempDir,
+		CacheSize:     conf.CacheSize,
+		BundleVersion: conf.BundleVersion,
+	}
+
+	if conf.Credentials != nil {
+		lp.SecretKey = conf.Credentials.WorkspaceSecret
+	}
+
+	if conf.Local != nil && conf.Local.EncryptionKey != "" {
+		encryptionKey, err := hex.DecodeString(conf.Local.EncryptionKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode encryption key: %w", err)
+		}
+
+		lp.EncryptionKey = encryptionKey
+	}
```

Comment:
This should be based on `conf.BundleVersion`. If it's v2, secret key doesn't matter and so on.